src/schemas/21-526EZ/schema.js

```diff
@@ -225,206 +282,233 @@ let schema = {
     treatments: {
       type: 'array',
+      maxItems: 100,
       items: {
         type: 'object',
         properties: {
-          treatment: { // Really hoping this extra nesting isn't necessary
-            type: 'object',
-            properties: {
-              treatmentCenterName: {
-                type: 'string'
-              },
-              // Can we make this in to a dateRange?
-              startTreatment: {
-                $ref: '#/definitions/date'
-              },
-              endTreatment: {
-                $ref: '#/definitions/date'
-              },
-              // Should this use a dropdown like address?
-              treatmentCenterCountry: {
-                type: 'string'
-              },
-              // Should this use a dropdown like address?
-              treatmentCenterState: {
-                type: 'string'
-              },
-              treatmentCenterCity: {
-                type: 'string'
-              },
-              treatmentCenterType: {
-                // Enum?
-                type: 'string'
-              }
-            }
+          treatmentCenterName: {
+            type: 'string',
+            maxLength: 100,
+            pattern: "([a-zA-Z0-9\-'.#]([a-zA-Z0-9\-'.# ])?)+$"
+          },
+          // Can we make this in to a dateRange?
+          startTreatmentDate: {
+            $ref: '#/definitions/datetime'
+          },
+          endTreatmentDate: {
+            $ref: '#/definitions/datetime'
+          },
+          // Should this use a dropdown like address?
+          treatmentCenterCountry: {
+            type: 'string'
+          },
+          // Should this use a dropdown like address?
+          treatmentCenterState: {
+            type: 'string',
+            pattern: '[a-zA-Z]{2}'
+          },
+          treatmentCenterCity: {
+            type: 'string',
+            maxLength: 100,
+            pattern: "([a-zA-Z0-9\-'.#]([a-zA-Z0-9\-'.# ])?)+$"
+          },
+          treatmentCenterType: {
+            type: 'string',
+            enum: ['VA_MEDICAL_CENTER', 'DOD_MTF']
           }
         }
       }
     },
     // Presumably, this should be an array...
     specialCircumstances: {
       type: 'array',
+      maxItems: 100,
       items: {
         type: 'object',
         properties: {
-          specialCircumstance: { // Really hoping this extra nesting isn't necessary
-            type: 'object',
-            properties: {
-              name: {
-                type: 'string'
-              },
-              code: {
-                type: 'string'
-              }
-            }
+          name: {
+            type: 'string'
+          },
+          code: {
+            type: 'string'
+          },
+          needed: {
+            type: 'boolean',
+            default: false
           }
         }
       }
     }
   },
-  // Any idea?
-  required: []
+  required: ['veteran', 'disabilities', 'serviceInformation']
```

Comment:
Probably not. I meant to check what we've done on other forms, but forgot. Do we normally leave out the `required` property altogether in the schema (requiring instead in the `formConfig`)?

Thanks for double-checking. :+1:

Reply:
They're different things. This required array will make it so that the array has to exist in the data, minItems will not. minItems will force there to be at least one item, if the array property exists in the data.

Reply:
Ahh, right. That makes sense. :+1: